msbuild/Xamarin.MacDev.Tasks/Tasks/FilterStaticFrameworks.cs

```diff
@@ -20,6 +20,29 @@ public class FilterStaticFrameworks : XamarinTask, ITaskCallback {
 		[Output]
 		public ITaskItem []? FrameworkToPublish { get; set; }
 
+		static string GetFrameworkExecutablePath (string frameworkPath)
+		{
+			if (!frameworkPath.EndsWith (".framework", StringComparison.OrdinalIgnoreCase) || !Directory.Exists (frameworkPath))
+				return frameworkPath;
+
+			// Try to read the CFBundleExecutable from Info.plist
+			var infoPlistPath = Path.Combine (frameworkPath, "Info.plist");
+			if (File.Exists (infoPlistPath)) {
+				try {
+					var plist = PDictionary.FromFile (infoPlistPath);
+					var bundleExecutable = plist?.GetCFBundleExecutable ();
+					if (!string.IsNullOrEmpty (bundleExecutable)) {
+						return Path.Combine (frameworkPath, bundleExecutable);
+					}
+				} catch {
+					// If reading the plist fails, fall back to the default behavior
+				}
```

Comment:
FromFile does not raise any exception, what exception are you trying to catch? This can me cleaner with a if statement no need for the try/catch.